### PR TITLE
Port FRC LabVIEW Programming from Screensteps

### PR DESCRIPTION
**Pull request by yellowjaguar5**
I think it looks okay, but it might need some linting. Let me know if you have any suggestions.

### CODE REVIEW

File: source/docs/software/labView/resources/local-and-global-variables-in-labview-for-frc.rst [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+You may create a local variable from the Structures palette on the block diagram as well.  When you have multiple local variables in one VI, you can left-click to choose which variable it is:
+
+.. image::images/local-and-global-variables-in-labview-for-frc/2.png
+
+Global variables are created slightly differently.  Add one to the block diagram from the Structures palette, and notice that when you double-click it, it opens a separate front panel.  This front panel does not have a block diagram, but you add as many entities to the front panel as you wish and save it as a \*.vi file:
+
+.. image::images/local-and-global-variables-in-labview-for-frc/3.png
+
+.. note:: Note: Be very careful to avoid race conditions when using local and global variables!  Essentially, make sure that you are not accidentally writing to the same variable in multiple locations without a way to know to which location it was last written.  For a more thorough explanation, see `this help document <http://zone.ni.com/reference/en-XX/help/371361L-01/lvconcepts/using_local_and_global/>`_
```
Comment (Daltz333) [inline source/docs/software/labView/resources/local-and-global-variables-in-labview-for-frc.rst]: Remove the redundant "Note" here. Not the directive, just the word at the beginning of the paragraph.

File: source/docs/software/labView/resources/local-and-global-variables-in-labview-for-frc.rst [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+How They are Used in the Default LabVIEW for FRC Robot Project
+--------------------------------------------------------------
+
+Global variables for “Enable Vision” and “Image Size” are written to during each iteration of the Robot Main VI…
+
+.. image::images/local-and-global-variables-in-labview-for-frc/4.png
+
+…And then read in each iteration of the Vision Processing VI:
```
Comment (Daltz333) [inline source/docs/software/labView/resources/local-and-global-variables-in-labview-for-frc.rst]:
```suggestion
… And then read in each iteration of the Vision Processing VI:
```

File: source/docs/software/labView/resources/making-a-one-shot-button-press.rst [hunk 3]
```diff
@@ -0,0 +1,38 @@
+Keyboard Navigation with the roboRIO
+====================================
+
+.. image::images/ni-logo.png
+
+When using the Joystick Get Values function, pushing a joystick button will cause the button to read TRUE until the button is released. This means that you will most likely read multiple TRUE values for each press. What if you want to read only one TRUE value each time the button is pressed? This is often called a "One-Shot Button". The following tutorial will show you how to create a subVI that you can drop into your Teleop.vi to do this.
+
+FIRSTly, create a new VI in the Support Code folder of your project.
```
Comment (Daltz333) [inline source/docs/software/labView/resources/making-a-one-shot-button-press.rst]:
```suggestion
First, create a new VI in the Support Code folder of your project.
```

File: source/docs/software/labView/resources/talon-srx-can.rst [hunk 4]
```diff
@@ -0,0 +1,8 @@
+Talon SRX CAN
+=============
+
+The Talon SRX motor controller is a CAN-enabled "smart motor controller" from Cross The Road Electronics/VEX Robotics. The Talon SRX can be controlled over the CAN bus or PWM interface. When using the CAN bus control, this device can take inputs from limit switches and potentiometers, encoders, or similar sensors in order to perform advanced control such as limiting or PID(F) closed loop control on the device.
+
+Extensive documentation about programming the Talon SRX in all three FRC languages can be found in the `Talon SRX Software Reference Manual on CTRE's Talon SRX product page <http://www.ctr-electronics.com/talon-srx.html#product_tabs_technical_resources>`_.
```
Comment (Daltz333) [inline source/docs/software/labView/resources/talon-srx-can.rst]: Convert this to an anonymous hyperlink, guidelines will be updated to reflect this change. 

```suggestion
Extensive documentation about programming the Talon SRX in all three FRC languages can be found in the `Talon SRX Software Reference Manual on CTRE's Talon SRX product page <http://www.ctr-electronics.com/talon-srx.html#product_tabs_technical_resources>`__.
```

File: source/docs/software/labView/resources/talon-srx-can.rst [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+
+Note: CAN Talon SRX has been removed from WPILib. See `this blog <http://www.firstinspires.org/robotics/frc/blog/2017-control-system-update>`_ for more info and find the CTRE Toolsuite installer here: http://www.ctr-electronics.com/control-system/hro.html#product_tabs_technical_resources
```
Comment (Daltz333) [inline source/docs/software/labView/resources/talon-srx-can.rst]:
```suggestion
Note: CAN Talon SRX has been removed from WPILib. See `this blog <http://www.firstinspires.org/robotics/frc/blog/2017-control-system-update>`__ for more info and find the CTRE Toolsuite installer `here <http://www.ctr-electronics.com/control-system/hro.html#product_tabs_technical_resources>`__
```

File: source/docs/software/labView/resources/using-the-compressor-in-labview.rst [hunk 6]
```diff
@@ -0,0 +1,35 @@
+Using the Compressor in LabVIEW
+===============================
+
+.. image::images/ni-logo.png
+
+This snippet shows how to set up your roboRIO project to use the Pneumatic Control Module (PCM).  The PCM automatically starts and stops the compressor when specific pressures are measured in the tank. In your roboRIO program, you will need to add the following VIs.
+
+For more information, check out the following links:
+
+https://rps01.usfirst.org/frc/manual/2015/2015FRCPneumaticsManual.pdf
```
Comment (Daltz333) [inline source/docs/software/labView/resources/using-the-compressor-in-labview.rst]: Don't just use the raw links, perhaps do something like?

FRC Pneumatics Manual

PCM User's Guide

Pneumatics Step by Step for the roboRIO

File: source/docs/software/labView/creating-robot-programs/creating-building-and-loading-your-benchtop-test-program.rst [hunk 7]
```diff
@@ -0,0 +1,56 @@
+Creating, Building and Loading your Benchtop Test Program
+=========================================================
+
+.. image::images/creating-building-and-loading-your-benchtop-test-program/ni-logo.png
```
Comment (Daltz333) [inline source/docs/software/labView/creating-robot-programs/creating-building-and-loading-your-benchtop-test-program.rst]: Put a space between the directive double colon and the hyperlink of the directive.

```suggestion
.. image:: images/creating-building-and-loading-your-benchtop-test-program/ni-logo.png
```

File: source/docs/software/labView/creating-robot-programs/creating-building-and-loading-your-benchtop-test-program.rst [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+.. note:: This document covers how to create, build and load an FRC LabVIEW program onto a roboRIO. Before beginning, make sure that you have installed LabVIEW for FRC and the FRC Driver Station and that you have configured and imaged your roboRIO as described in the Getting Started with the Control System section.
+
+.. todo:: Add link to the Getting Started with the Control System section.
```
Comment (Daltz333) [inline source/docs/software/labView/creating-robot-programs/creating-building-and-loading-your-benchtop-test-program.rst]:
```suggestion
```

File: source/docs/software/labView/creating-robot-programs/creating-building-and-loading-your-benchtop-test-program.rst [hunk 9]
```diff
@@ -0,0 +1,56 @@
+Creating, Building and Loading your Benchtop Test Program
+=========================================================
+
+.. image::images/creating-building-and-loading-your-benchtop-test-program/ni-logo.png
+
+.. note:: This document covers how to create, build and load an FRC LabVIEW program onto a roboRIO. Before beginning, make sure that you have installed LabVIEW for FRC and the FRC Driver Station and that you have configured and imaged your roboRIO as described in the Getting Started with the Control System section.
```
Comment (Daltz333) [inline source/docs/software/labView/creating-robot-programs/creating-building-and-loading-your-benchtop-test-program.rst]:
```suggestion
.. note:: This document covers how to create, build and load an FRC LabVIEW program onto a roboRIO. Before beginning, make sure that you have installed LabVIEW for FRC and the FRC Driver Station and that you have configured and imaged your roboRIO as described in the :ref:`Getting Started with the Control System section <docs/software/getting-started/index:Getting Started>`.
```

File: source/docs/software/labView/index.rst [hunk 10]
```diff
@@ -0,0 +1,35 @@
+FRC LabVIEW Programming
+=======================
+
+.. toctree::
+   :maxdepth: 1
+   :caption: Setting up the Development Environment
+
+   installing-labview-for-frc-2019-labview-only
+
+.. todo:: Add link to ``Installing the FRC Update Suite (All Languages)``
```
Comment (Daltz333) [inline source/docs/software/labView/index.rst]: Remove, no need to put a link here.

```suggestion
```

File: source/docs/software/labView/resources/how-to-use-joystick-buttons-to-control-motors-or-solenoids.rst [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+Scenario 3
+-----------
+
+I want to control a solenoid with my joystick buttons.
+
+By now, we are familiar with how the joystick outputs the buttons in an array of booleans.  We need to index this array to get the button we are interested in, and wire this boolean to a select node.  Since the "Solenoid Set.vi" requires a Enum as an input, the easiest way to get the enum is to right click the "Value" input of the "Solenoid Set.vi" and select "Create Constant".  Duplicate this constant and wire one copy to the True terminal and one to the False terminal of the select node.  Then wire the output of the select node to the "Value" input of the solenoid VI.
+
+.. image::images/how-to-use-joystick-buttons-to-control-motors-or-solenoids/3.png
+
+Hope this helps!  Feel free to comment below.
```
Comment (Daltz333) [inline source/docs/software/labView/resources/how-to-use-joystick-buttons-to-control-motors-or-solenoids.rst]: Remove

```suggestion
```

**Comment by yellowjaguar5** [comment]
@Daltz333 I think I addressed all your requested changes and I (finally) figured out how to build sphynx locally, so this should go faster next time. Please let me know if you have any more suggestions.

File: source/docs/software/labview/resources/talon-srx-can.rst [hunk 12]
```diff
@@ -0,0 +1,8 @@
+Talon SRX CAN
+=============
+
+The Talon SRX motor controller is a CAN-enabled "smart motor controller" from Cross The Road Electronics/VEX Robotics. The Talon SRX can be controlled over the CAN bus or PWM interface. When using the CAN bus control, this device can take inputs from limit switches and potentiometers, encoders, or similar sensors in order to perform advanced control such as limiting or PID(F) closed loop control on the device.
+
+Extensive documentation about programming the Talon SRX in all three FRC languages can be found in the `Talon SRX Software Reference Manual on CTRE's Talon SRX product page <http://www.ctr-electronics.com/talon-srx.html#product_tabs_technical_resources>`__.
+
+Note: CAN Talon SRX has been removed from WPILib. See `this blog <http://www.firstinspires.org/robotics/frc/blog/2017-control-system-update>`__ for more info and find the CTRE Toolsuite installer `here <http://www.ctr-electronics.com/control-system/hro.html#product_tabs_technical_resources>`__
```
Comment (Daltz333) [inline source/docs/software/labview/resources/talon-srx-can.rst]:
```suggestion
.. note:: CAN Talon SRX has been removed from WPILib. See `this blog <http://www.firstinspires.org/robotics/frc/blog/2017-control-system-update>`__ for more info and find the CTRE Toolsuite installer `here <http://www.ctr-electronics.com/control-system/hro.html#product_tabs_technical_resources>`__
```

File: source/docs/software/labview/resources/how-to-use-joystick-buttons-to-control-motors-or-solenoids.rst [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+No matter what the configuration, you'll need to add one, two, or more (if you're really excited) joysticks to the "Begin.vi".  The first example uses 2 joysticks and the others only use one.  Give each one a unique name so we can use it in other places, like the snippet below.  I named them "LeftStick" and "RightStick" because they are on the left and right sides of my desk.  If your joysticks are already configured, great! You can skip this step.
+
+.. image:: images/how-to-use-joystick-buttons-to-control-motors-or-solenoids/setup.png
+
+The rest of the code in this document will be placed in the "Teleop.VI" This is where we will be programming our joystick buttons to control different aspects of our motors or solenoids.
+
+Scenario 1
+----------
+
+I want a motor to move one way when I press one button and the other way when I press a different button.
```
Comment (Daltz333) [inline source/docs/software/labview/resources/how-to-use-joystick-buttons-to-control-motors-or-solenoids.rst]: Could you surround the scenarios in quotation marks to add emphasis?

```suggestion
"I want a motor to move one way when I press one button and the other way when I press a different button."
```

File: source/docs/software/labview/resources/how-to-use-joystick-buttons-to-control-motors-or-solenoids.rst [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+
+Each case corresponds to a binary representation of the values in the array.  In this example, each case corresponds to a one-button combination.  We added six cases: 0 (all buttons off), 1 (button 0 on), 2 (button 1 on), 4 (button 2 on), 8 (button 3 on), and 16 (button 4 on).  Notice we skipped value 3.  3 would correspond to buttons 0 and 1 pressed at the same time.  We did not define this in our requirements so we'll let the default case handle it.
+
+It might be helpful to review the LabVIEW 2014 Case Structure Help document here:
+
+http://zone.ni.com/reference/en-XX/help/371361L-01/glang/case_structure/
+
+There are also 3 Community Tutorials on case structures here:
+
+http://forums.ni.com/docs/DOC-11621
```
Comment (Daltz333) [inline source/docs/software/labview/resources/how-to-use-joystick-buttons-to-control-motors-or-solenoids.rst]:
```suggestion
https://forums.ni.com/docs/DOC-11621
```

File: source/docs/software/labview/resources/how-to-use-joystick-buttons-to-control-motors-or-solenoids.rst [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+- button 4 - move forward at 1/4 speed
+- trigger - full speed ahead! (forward at full speed)
+
+We would then take the boolean array from the "JoystickGetValues.vi" and wire it to a "Boolean Array to Number" node (Numeric Palette-Conversion Pallette).  This converts the boolean array to a number that we can use.  Wire this numeric to a case structure.
+
+Each case corresponds to a binary representation of the values in the array.  In this example, each case corresponds to a one-button combination.  We added six cases: 0 (all buttons off), 1 (button 0 on), 2 (button 1 on), 4 (button 2 on), 8 (button 3 on), and 16 (button 4 on).  Notice we skipped value 3.  3 would correspond to buttons 0 and 1 pressed at the same time.  We did not define this in our requirements so we'll let the default case handle it.
+
+It might be helpful to review the LabVIEW 2014 Case Structure Help document here:
+
+http://zone.ni.com/reference/en-XX/help/371361L-01/glang/case_structure/
```
Comment (Daltz333) [inline source/docs/software/labview/resources/how-to-use-joystick-buttons-to-control-motors-or-solenoids.rst]:
```suggestion
https://zone.ni.com/reference/en-XX/help/371361L-01/glang/case_structure/
```

File: source/docs/software/labview/resources/how-to-use-joystick-buttons-to-control-motors-or-solenoids.rst [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+
+There are also 3 Community Tutorials on case structures here:
+
+http://forums.ni.com/docs/DOC-11621
+
+http://forums.ni.com/docs/DOC-11622
```
Comment (Daltz333) [inline source/docs/software/labview/resources/how-to-use-joystick-buttons-to-control-motors-or-solenoids.rst]:
```suggestion
https://forums.ni.com/docs/DOC-11622
```

File: source/docs/software/labview/resources/how-to-use-joystick-buttons-to-control-motors-or-solenoids.rst [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+
+http://forums.ni.com/docs/DOC-11623
```
Comment (Daltz333) [inline source/docs/software/labview/resources/how-to-use-joystick-buttons-to-control-motors-or-solenoids.rst]:
```suggestion
https://forums.ni.com/docs/DOC-11623
```

File: source/docs/software/labview/resources/how-to-use-joystick-buttons-to-control-motors-or-solenoids.rst [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+Scenario 3
+-----------
+
+I want to control a solenoid with my joystick buttons.
+
+By now, we are familiar with how the joystick outputs the buttons in an array of booleans.  We need to index this array to get the button we are interested in, and wire this boolean to a select node.  Since the "Solenoid Set.vi" requires a Enum as an input, the easiest way to get the enum is to right click the "Value" input of the "Solenoid Set.vi" and select "Create Constant".  Duplicate this constant and wire one copy to the True terminal and one to the False terminal of the select node.  Then wire the output of the select node to the "Value" input of the solenoid VI.
+
+.. image:: images/how-to-use-joystick-buttons-to-control-motors-or-solenoids/3.png
+
+
```
Comment (Daltz333) [inline source/docs/software/labview/resources/how-to-use-joystick-buttons-to-control-motors-or-solenoids.rst]: Extra space here

```suggestion
```

File: source/docs/software/labview/resources/how-to-use-joystick-buttons-to-control-motors-or-solenoids.rst (hunk 13 above)
Comment (yellowjaguar5) [reply source/docs/software/labview/resources/how-to-use-joystick-buttons-to-control-motors-or-solenoids.rst]: Sure. I think bold might also make it clearer.
```suggestion
**"I want a motor to move one way when I press one button and the other way when I press a different button."**
```

**Pull request merged**